packages/blocks/src/api/constants.js

```diff
@@ -36,6 +36,11 @@ export const __EXPERIMENTAL_STYLE_PROPERTY = {
 		requiresOptOut: true,
 		useEngine: true,
 	},
+	backgroundImage: {
+		value: [ 'background', 'backgroundImage' ],
+		support: [ 'background', 'backgroundImage' ],
+		useEngine: false,
```

Comment:
Possibly temporary, possibly not.

Bypass style engine because it's expecting `url` and `source` properties.

https://github.com/WordPress/gutenberg/blob/f7bb5df897cabc1a34e678696382ba90f5f24890/packages/style-engine/src/styles/background/index.ts#L19-L19

But I'm not yet convinced these need to be replicated in theme.json, which should support at first an image path string value for `background.backgroundImage` (?), and potentially ref paths in blocks, e.g., `"backgroundImage": { "ref": "styles.background.backgroundImage" }`

Options:

1. Convert `background.backgroundImage (string)` to `array()` with `url` and `source` properties in the backend after processing theme.json
2. Imply in all contexts that URL-like strings coming from `styles.background.backgroundImage` are equal to `{ source: 'file', url: styles.background.backgroundImage }`
3. ...

Reply:
I think we might want to support an object for `background.backgroundImage` in each context (block attributes and theme.json) in order to allow `source` to be set to things other than `file` (e.g. `featuredImage`, `theme` if we want to prepend the theme's directory to the URL, etc). But, for ease of use, it'd likely be good have it support a simple string, too.

How bad would it be to allow the property to be a string or an object?

Reply:
> source to be set to things other than file (e.g. featuredImage, theme if we want to prepend the theme's directory to the URL, etc)

Thanks for the explainer @andrewserong  I couldn't find any background on why `source` was there.

I suppose having `source` as a context marker would be handy to add extra functionality.

> How bad would it be to allow the property to be a string or an object?

In that case, my opinion is that we could support both in theme.json. 

Where `"backgroundImage": "http://some_url/image.gif" is the equivalent of `{ source: 'file', url: 'http://some_url/image.gif' }`

Thank you!

Reply:
> Where "backgroundImage": "http://some_url/image.gif" is the equivalent of { source: 'file', url: 'http://some_url/image.gif' }

Sounds great to me!

Reply:
Interesting... the style engine expects string values and does nothing with them

https://github.com/WordPress/gutenberg/blob/d3a74ac1feb9ed6d230118db2f45ce90ab725843/packages/style-engine/class-wp-style-engine.php#L611-L611

I just realized this is a good idea in case uses want something like:

```css
background-image: linear-gradient(
    to bottom,
    rgb(255 255 0 / 50%),
    rgb(0 0 255 / 50%)
  ), url("catfront.png");
```

Example taken from https://developer.mozilla.org/en-US/docs/Web/CSS/background-image

I've updated the branch to wave strings through like this.